src/twister2/log_file/build_log_file.py

```diff
@@ -0,0 +1,27 @@
+from __future__ import annotations
+
+import logging
+from pathlib import Path
+
+from twister2.log_file.log_file_abstract import LogFileAbstract
+
+logger = logging.getLogger(__name__)
+
+
+class BuildLogFile(LogFileAbstract):
+    """Save logs from the building."""
+    name = 'build'
+
+    def __init__(self, build_dir: str | Path) -> None:
+        super().__init__(build_dir=build_dir, name=self.name)
+
+    def handle(self, data: str | bytes) -> None:
+        """
+        :param data: output from building
+        """
+        if data:
+            data = data.decode(encoding=self.default_encoding) if type(data) is bytes else data
+            with open(file=self.filename, mode='w+', encoding=self.default_encoding) as log_file:
```

Comment:
Probably you should use append mode (`"a"`), instead of write (`"w+"`). Now when I run hello_world sample, then in `build.log` file I can see only part responsible for Ninja generator - the part responsible for CMake was overwritten. During building process `_run_command_in_subprocess` are called twice times (first time by `run_cmake_stage` method, second one by `run_build_generator` method)

To reproduce:
call Twister v2 by command:
```
pytest -vvs --platform=native_posix samples/hello_world 
```
call Twister v1 by command:
```
./scripts/twister -vv --platform=native_posix -T samples/hello_world
```
Compare generated build logs by for example `meld` tool (or `diff`):
```
meld twister-out/native_posix/samples/hello_world/sample.basic.helloworld/build.log twister-out.1/native_posix/samples/hello_world/sample.basic.helloworld/build.log
```
WARNING - some minor differences (like order of built elements) are acceptable

Reply:
Done